src/java.base/share/classes/java/lang/reflect/code/bytecode/BytecodeGenerator.java

```diff
@@ -691,11 +740,51 @@ private void generateBody(Body body) {
                         processOperands(op, isLastOpResultOnStack);
                         cob.checkcast(((JavaType) op.type()).toNominalDescriptor());
                     }
+                    case LambdaOp op -> {
+                        JavaType intfType = (JavaType)op.functionalInterface();
+                        try {
+                            Class<?> intfClass = intfType.resolve(lookup);
+                            for (Value cv : op.capturedValues()) {
+                                load(cv);
+                            }
+                            MethodTypeDesc mtd = MethodRef.toNominalDescriptor(op.invokableType());
+                            ClassDesc[] captureTypes = op.capturedValues().stream().map(Value::type).map(BytecodeGenerator::toClassDesc).toArray(ClassDesc[]::new);
+                            if (Quotable.class.isAssignableFrom(intfClass)) {
+                                // @@@ double the captured values to enable LambdaMetafactory.FLAG_QUOTABLE
+                                for (Value cv : op.capturedValues()) {
+                                    load(cv);
+                                }
+                                cob.invokedynamic(DynamicCallSiteDesc.of(
+                                        DMHD_LAMBDA_METAFACTORY,
+                                        funcIntfMethodName(intfClass),
+                                        // @@@ double the descriptor parameters to enable LambdaMetafactory.FLAG_QUOTABLE
+                                        MethodTypeDesc.of(intfType.toNominalDescriptor(), Stream.concat(Stream.of(captureTypes), Stream.of(captureTypes)).toList()),
+                                        mtd,
+                                        MethodHandleDesc.ofMethod(DirectMethodHandleDesc.Kind.STATIC, className, "lambda$" + lambdaSink.size(), mtd.insertParameterTypes(0, captureTypes)),
+                                        mtd,
+                                        LambdaMetafactory.FLAG_QUOTABLE,
+                                        MethodHandleDesc.ofField(DirectMethodHandleDesc.Kind.STATIC_GETTER, className, "lambda$" + lambdaSink.size() + "$op", CD_String)));
+                                quotable.set(lambdaSink.size());
+                            } else {
+                                cob.invokedynamic(DynamicCallSiteDesc.of(
+                                        DMHD_LAMBDA_METAFACTORY,
```

Comment:
This should use the `metafactory` bootstrap rather than `altMetafactory`.

Reply:
Yep, split to use `metafactory`, thanks.